client/js/components/EditProfileForm/EditProfileForm.js

```diff
@@ -0,0 +1,199 @@
+import React from "react";
+import { Form, Field } from "react-final-form";
+import {
+  View,
+  Image,
+  TouchableOpacity,
+  TextInput,
+  Text,
+  Alert
+} from "react-native";
+import styles from "./styles";
+import PropTypes from "prop-types";
+
+class EditProfileForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newPassword: "",
+      verifyPassword: "",
+      firstname: this.props.data ? this.props.data.firstname : "First name",
+      lastname: this.props.data ? this.props.data.lastname : "Last name",
+      email: this.props.data ? this.props.data.email : "Email"
```

Comment:
we never update state or create state using props in the constructor. Use the lifecycle methods.